File: v6/sprefa-extract/tests/fixtures/df_loops/sample.kt
class Cell(val value: Int)

fun allocating(rows: List<Int>, cols: List<Int>): List<Cell> {
    val out = mutableListOf<Cell>()
    for (row in rows) {
        for (col in cols) {
            val cell = Cell(add(row, col))
            out.add(cell)
        }
    }
    return out
}

fun plain(limit: Int): Int {
    var total = 0
    while (total < limit) {
        total = add(total, 1)
    }
    return add(total, 0)
}

fun add(left: Int, right: Int): Int {
    return left + right
}

File: v6/sprefa-extract/tests/fixtures/df_loops/sample.rs
struct Cell {
    value: i64,
}

fn allocating(rows: &[i64], cols: &[i64]) -> Vec<Cell> {
    let mut out = Vec::new();
    for row in rows {
        for col in cols {
            let cell = Cell {
                value: add(*row, *col),
            };
            out.push(cell);
        }
    }
    out
}

fn plain(limit: i64) -> i64 {
    let mut total = 0;
    while total < limit {
        total = add(total, 1);
    }
    add(total, 0)
}

fn add(left: i64, right: i64) -> i64 {
    left + right
}
